### DISCUSSION TITLE
Importing CDN for ImageMagick webassembly on a static website

### DISCUSSION THREAD

**Discussion by RedMonkeyGone** [ImageMagick/ImageMagick#7736] [Development] [answered]
### ImageMagick version

magick-wasm 0.0.31

### Operating system, version and so on

Nginx Hugo Webserver

### Description

As part of my website, I've been trying to have a free image converter, making it easy for me and other users to convert images. It's a hugo static website.

As part of this, I'm trying to import magick-wasm but I can't get it to work.

With my current code, I'm given this error

` index.js:2995 failed to asynchronously prepare wasm: CompileError: WebAssembly.instantiate(): BufferSource argument is empty
(anonymous) @   index.js:2995
Promise.then        
Sn  @   index.js:2994
Nr  @   index.js:3004
Hr  @   index.js:3022
(anonymous) @   index.js:5706
(anonymous) @   index.js:7462
loader  @   index.js:7456
_initialize @   index.js:7477
oo  @   index.js:7521
initializeMagickConverter   @   work/:234
(anonymous) @   work/:244 ` 

` index.js:2954 Aborted(CompileError: WebAssembly.instantiate(): BufferSource argument is empty)
de  @   index.js:2954
(anonymous) @   index.js:2995
Promise.then        
Sn  @   index.js:2994
Nr  @   index.js:3004
Hr  @   index.js:3022
(anonymous) @   index.js:5706
(anonymous) @   index.js:7462
loader  @   index.js:7456
_initialize @   index.js:7477
oo  @   index.js:7521
initializeMagickConverter   @   work/:234
(anonymous) @   work/:244`

` index.js:2955 Uncaught (in promise) RuntimeError: Aborted(CompileError: WebAssembly.instantiate(): BufferSource argument is empty). Build with -sASSERTIONS for more info.
    at de (index.js:2955:15)
    at index.js:2995:60
de  @   index.js:2955
(anonymous) @   index.js:2995
Promise.then        
(anonymous) @   index.js:7462
loader  @   index.js:7456
_initialize @   index.js:7477
oo  @   index.js:7521
initializeMagickConverter   @   work/:234
(anonymous) @   work/:244 `

And no matter what I move around I'm not sure what the correct import should look like to use the cdn for magick wasm correctly.

[testconvert.txt](https://github.com/user-attachments/files/17603705/testconvert.txt)
This is the simplified version of my code you can run as an html file.

Appreciate any help you could provide. Thank you.

**Comment by dlemstra** [comment, marked as answer]
Why do you think that this works?

```js
magick = await initializeImageMagick({
    locateFile: (file) => `https://cdn.jsdelivr.net/npm/@imagemagick/magick-wasm@0.0.31/dist/${file}`,
});
```

That first argument should be `URL | ByteArray | WebAssembly.Module`. You should probably use `https://cdn.jsdelivr.net/npm/@imagemagick/magick-wasm@0.0.31/dist/magick.wasm` instead?

**Reply by RedMonkeyGone** [reply]
really appreciate you getting back to me.

Realised I'm making something no one will use. Let's put an end to this here.
